extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/UseAsQualifierScrutinizer.java

```diff
@@ -0,0 +1,84 @@
+package org.openrefine.wikidata.qa.scrutinizers;
+
+import org.openrefine.wikidata.qa.QAWarning;
+import org.openrefine.wikidata.updates.ItemUpdate;
+import org.wikidata.wdtk.datamodel.helpers.Datamodel;
+import org.wikidata.wdtk.datamodel.interfaces.PropertyIdValue;
+import org.wikidata.wdtk.datamodel.interfaces.Snak;
+import org.wikidata.wdtk.datamodel.interfaces.SnakGroup;
+import org.wikidata.wdtk.datamodel.interfaces.Statement;
+import org.wikidata.wdtk.datamodel.interfaces.Value;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class UseAsQualifierScrutinizer extends EditScrutinizer {
+
+    public static final String type = "values-should-not-be-used-as-qualifier";
+    public static String ONE_OF_QUALIFIER_VALUE_PROPERTY_QID = "Q52712340";
+    public static String QUALIFIER_PROPERTY_PID = "P2306";
+    public static String ITEM_OF_PROPERTY_CONSTRAINT_PID = "P2305";
+
+    class UseAsQualifierConstraint {
+        final PropertyIdValue allowedQualifierPid;
+        final List<Value> itemList;
+        UseAsQualifierConstraint(Statement statement) {
+            List<SnakGroup> specs = statement.getClaim().getQualifiers();
+            PropertyIdValue pid = null;
+            this.itemList = new ArrayList<>();
+            for(SnakGroup group : specs) {
+                for (Snak snak : group.getSnaks()) {
+                    if (group.getProperty().getId().equals(QUALIFIER_PROPERTY_PID)){
+                        pid = (PropertyIdValue) snak.getValue();
+                    }
+                    if (group.getProperty().getId().equals(ITEM_OF_PROPERTY_CONSTRAINT_PID)){
+                        this.itemList.add(snak.getValue());
+                    }
+                }
+            }
+            this.allowedQualifierPid = pid;
+        }
+    }
+
+    @Override
+    public void scrutinize(ItemUpdate update) {
+        for (Statement statement : update.getAddedStatements()) {
+            PropertyIdValue pid = statement.getClaim().getMainSnak().getPropertyId();
+            Map<PropertyIdValue, List<Value>> qualifiersMap = new HashMap<>();
+            List<SnakGroup> specs = statement.getClaim().getQualifiers();
```

Comment:
Why call this variable `specs`? Those are the qualifiers you are inspecting, not the specifications of a constraint.

Reply:
Copy-paste error. Sorry. :(